### PR TITLE
bugfix: fix known problems of docker image

### LINKED ISSUE #6358: jib-maven-plugin copy resource failed
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

image
https://github.com/apache/incubator-seata/actions/runs/7962346784/job/21735611187

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by xingfudeshi**
- [x] I have registered the PR [changes](../changes).

### Ⅱ. Does this pull request fix one issue?

fixes #6358 
also fixed the following problems

under:openjdk:8-jre-slim

Pasted Graphic 1

under:openjdk:11-jre-stretch
Pasted Graphic 2

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
Tested passed in the following version of the base image:
8
9
11
17

### CODE REVIEW

File: server/pom.xml [hunk 1]
```diff
@@ -393,10 +393,6 @@
                     </container>
                     <extraDirectories>
                         <paths>
-                            <path>
```
Comment (slievrly) [inline server/pom.xml:396]: Need to preserve the empty folder for jdbc.

Comment (xingfudeshi) [reply server/pom.xml:396]: OK

Comment (xingfudeshi) [reply server/pom.xml:396]: image
done.

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**